jobdescription.py

```diff
@@ -0,0 +1,517 @@
+
+    def set_description_parameter(self, key, value):
+        if self.__holder is not None:
+            if key in self.__holder:
+                self.__holder[key] = value
+                return True
+
+            if key in self.__input_file_keys:
+                err = "Key JobDescription.%s is read-only\n" % key
+                if key == 'inFiles':
+                    err += "Use JobDescription.input_files to manipulate input files"
+                else:
+                    err += "Use JobDescription.input_files[][%s] to set up this parameter in files description" % self.__input_file_keys[key]
+                raise AttributeError(err)
+
+            if key in self.__output_file_keys:
+                err = "Key JobDescription.%s is read-only\n" % key
+                if key == 'outFiles':
+                    err += "Use JobDescription.output_files to manipulate output files"
+                else:
+                    err += "Use JobDescription.output_files[][%s] to set up this parameter in files description" % self.__output_file_keys[key]
+                raise AttributeError(err)
+
+            snake_key = camel_to_snake(key)
+            if snake_key in self.__key_aliases_snake:
+                log.warning("Better to use %s to access and manipulate this value" % self.__key_aliases_snake[snake_key])
+                self.__holder[self.__key_aliases_snake[snake_key]] = parse_value(value)
+
+            if key in self.__soft_key_aliases:
+                return self.set_description_parameter(self.__soft_key_aliases[key], value)
+
+        return False
+
+    def __getattr__(self, key):
+        """
+        Reflection of description values into Job instance properties if they are not shadowed.
+        If there is no own property with corresponding name, the value of Description is used.
+        Params and return described in __getattr__ interface.
+        """
+        try:
+            return object.__getattribute__(self, key)
+        except AttributeError:
+            return self.get_description_parameter(key)
+
+    def __setattr__(self, key, value):
+        """
+        Reflection of description values into Job instance properties if they are not shadowed.
+        If there is no own property with corresponding name, the value of Description is set.
+        Params and return described in __setattr__ interface.
+        """
+        try:
+            object.__getattribute__(self, key)
+            return object.__setattr__(self, key, value)
+        except AttributeError:
+            if not self.set_description_parameter(key, value):
+                return object.__setattr__(self, key, value)
+
+
+if __name__ == "__main__":
+    import sys
+    logging.basicConfig()
+    log.setLevel(logging.DEBUG)
+
+    jd = JobDescription()
+    with open(sys.argv[1], "r") as f:
+        contents = f.read()
+
+    jd.load(contents)
+
+    log.debug(jd.id)
+    log.debug(jd.command)
```

Comment:
it's a bit confusing with the renaming of variables - is it possible to use both the renamed variable as well as the original variable? i.e. is jd.id the same as jd.PandaID? jd.command is a renamed variable, from transformation, but below you use original variable names like scopeOut etc.

Reply:
Yeah, it is possible, and it's a part of the plan. I tried to point it out in the e-mail.
As I proposed these renamings nearly one and a half years ago and I know for sure what's gonna happen on their introduction to others.
Minipilot already uses those everywhere and I just needed to swap the straight usage by this class mixin.
But yes, Pilot 2 uses old titles and when I tried to introduce my revision (along with other changes)   a year ago, Mario and other developers who were working back then on the P2, were too angry and they didn't appreciate any of my changes, they were too radical to get them in one go. Back then I thought even to make my own fork of p2 and introduce it as a concurrent solution.
So, here is the test piece, that shows the possibility to get the variables by their old names, that will help to introduce this change faster.
And then, when introduced, I can put a warning showing where the old name calls happen, to easily switch them to new ones.

Reply:
I think we should agree on the new names and then not use the old ones or we are bound for trouble. I don't think we should allow the old names at all. it is confusing. let's only use the new ones. if we're not happy with a particular name we can change it.

Reply:
Sure. The old ones will remain only while transferring. But they _will_ remain, or I will again get all the anger of Mario and others.
For the namings, why I chose each of those — I can explain. For example, command instead of transformation is used because thus PanDA will show it's compatibility with any tasks not only Atlas ones.

Reply:
If you wish, I'll introduce the warning right away, it's an easy one to implement

Reply:
yes that's a good idea. please add a warning. maybe then come up with a better name than 'command', perhaps 'script'

Reply:
OK, let it be script. And script parameters then.

Reply:
right. it's good as well.

Reply:
Done